components/Home/Bonding/TokenManagement/WsolWithdraw.jsx

```diff
@@ -0,0 +1,223 @@
+import { useWsolWithdraw } from './hooks/useWsolWithdraw';
+import { DEFAULT_SLIPPAGE, slippageValidator } from './utils';
+
+export const WsolWithDraw = () => {
+  const [form] = Form.useForm();
+  const [isEstimating, setIsEstimating] = useState(false);
+  const [isWithdrawing, setIsWithdrawing] = useState(false);
+  const [maxAmount, setMaxAmount] = useState(0);
+
+  const { isSvmWalletConnected } = useSvmConnectivity();
+  const {
+    withdrawDecreaseLiquidityQuote: fn,
+    withdrawTransformedQuote,
+    withdraw,
+    getMaxAmount,
+  } = useWsolWithdraw();
+  const getDecreaseLiquidityQuote = pDebounce(fn, 500);
+
+  // initially, set default slippage value
+  useEffect(() => {
+    form.setFieldsValue({ slippage: DEFAULT_SLIPPAGE });
+  }, [form]);
+
+  const updateMaxAmount = useCallback(async () => {
+    const setMaxAmountFn = async () => {
+      const tempAmount = await getMaxAmount();
+      if (!isNil(tempAmount)) {
+        setMaxAmount(tempAmount / LAMPORTS_PER_SOL);
+      }
+    };
+
+    if (isSvmWalletConnected) setMaxAmountFn();
+  }, [isSvmWalletConnected, getMaxAmount]);
+
+  // set max amount on initial load
+  useEffect(() => updateMaxAmount(), [updateMaxAmount]);
+
+  const onAmountAndSlippageChange = useCallback(async () => {
+    const amount = form.getFieldValue('amount');
+    const slippage = form.getFieldValue('slippage');
+
+    if (!isNumber(amount) || !isNumber(slippage)) return;
+
+    try {
+      setIsEstimating(true);
+
+      const actualAmount = amount * LAMPORTS_PER_SOL;
+      const quote = await getDecreaseLiquidityQuote({
+        amount: actualAmount,
+        slippage,
+      });
+      const transformedQuote = await withdrawTransformedQuote(quote);
+
+      // update olas and wsol value
+      form.setFieldsValue({
+        olas: transformedQuote?.olasMin || null,
+        wsol: transformedQuote?.wsolMin || null,
+      });
+    } catch (error) {
+      notifyError('Failed to get estimated quote');
+      console.error(error);
+    } finally {
+      setIsEstimating(false);
+    }
+  }, [form, getDecreaseLiquidityQuote, withdrawTransformedQuote]);
+
+  const onMaxClick = useCallback(() => {
+    form.setFieldsValue({ amount: maxAmount });
+    onAmountAndSlippageChange();
+  }, [form, maxAmount, onAmountAndSlippageChange]);
+
+  const handleWithdraw = useCallback(async () => {
+    if (!isSvmWalletConnected) {
+      notifyError('Please connect your wallet');
+      return;
+    }
+
+    const amount = form.getFieldValue('amount');
+    const slippage = form.getFieldValue('slippage');
+    if (amount > maxAmount) {
+      notifyError('Amount exceeds the maximum limit');
+      return;
+    }
+
+    try {
+      setIsWithdrawing(true);
+
+      const actualAmount = amount * LAMPORTS_PER_SOL;
+      await withdraw({ amount: actualAmount, slippage });
+
+      // reset form fields after successful withdraw
+      form.setFieldsValue({
```

Comment:
if you use initialValues you can simply reset form by calling form.resetFields()